#! /usr/bin/env python


PACKAGE='tools'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        Name          		Type     Level Description			Default Min	Max
gen.add("front_pitch", 		double_t,	0,  "front_pitch", 		12.0,  -180,  180)
gen.add("front_tx", 		double_t,	0,  "front_tx", 		0,  -10,  10)
gen.add("front_ty", 		double_t,	0,  "front_ty", 		0,  -10,  10)
gen.add("front_tz", 		double_t,	0,  "front_tz", 		1.9,  -10,  10)
gen.add("rear_roll", 		double_t,	0,  "rear_roll", 		-22.0,  -180,  180)
gen.add("rear_pitch", 		double_t,	0,  "rear_pitch", 		15.5,  -180,  180)
gen.add("rear_yaw", 		double_t,	0,  "rear_yaw", 		90.0,  -180,  180)
gen.add("rear_tx", 			double_t,	0,  "rear_tx", 			0.2,  -10,  10)
gen.add("rear_ty", 			double_t,	0,  "rear_ty", 			-1.0,  -10,  10)
gen.add("rear_tz", 			double_t,	0,  "rear_tz", 			1.7,  -10,  10)


exit(gen.generate(PACKAGE, "lidar_calibration", "lidar_calibration"))
